docs/CNDP_README.md

```diff
@@ -0,0 +1,323 @@
+ # "sim" uses Source() modules to simulate traffic generation
+ # "cndp" use kernel AF-XDP. It supports ZC and XDP offload if driver and NIC supports it. It's tested on Intel 800 series n/w adapter.
+-mode="dpdk"
+-#mode="cndp"
++#mode="dpdk"
++mode="cndp"
+ #mode="af_xdp"
+ #mode="af_packet"
+ #mode="sim"
+```
+
+4) Modify [cndp_upf_1worker.jsonc](./conf/cndp_upf_1worker.jsonc) file lports section to update the access and core netdev interface name and required queue id.
+
+5) Modify the script [docker_setup.sh](./scripts/docker_setup.sh) and update the access and core interface names (s1u, sgi), access/core interface mac addresses and neighbor gateway interfaces mac addresses. This should match the access/core netdev interface name used in jsonc file in previous step. In our example test setup, neighbor mac address (n-s1u, n-sgi) corresponds to access/core interfaces used by packet generator in system 2 to send/receive n/w packets. Update following values based on your system configuration.
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 7aff6a6..09d640b 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -24,7 +24,7 @@ mode="dpdk"
+ # Gateway interface(s)
+ #
+ # In the order of ("s1u" "sgi")
+-ifaces=("ens803f2" "ens803f3")
++ifaces=("enp134s0" "enp136s0")
+
+ # Static IP addresses of gateway interface(s) in cidr format
+ #
+@@ -34,7 +34,7 @@ ipaddrs=(198.18.0.1/30 198.19.0.1/30)
+ # MAC addresses of gateway interface(s)
+ #
+ # In the order of (s1u sgi)
+-macaddrs=(9e:b2:d3:34:ab:27 c2:9c:55:d4:8a:f6)
++macaddrs=(40:a6:b7:78:3f:ec 40:a6:b7:78:3f:e8)
+
+ # Static IP addresses of the neighbors of gateway interface(s)
+ #
+@@ -44,7 +44,7 @@ nhipaddrs=(198.18.0.2 198.19.0.2)
+ # Static MAC addresses of the neighbors of gateway interface(s)
+ #
+ # In the order of (n-s1u n-sgi)
+-nhmacaddrs=(22:53:7a:15:58:50 22:53:7a:15:58:50)
++nhmacaddrs=(40:a6:b7:78:3f:bc 40:a6:b7:78:3f:b8)
+```
+
+6) Modify the script [docker_setup.sh](./scripts/docker_setup.sh) and update the function `move_ifaces()` in condition `if [ "$mode" == 'cndp' ]` . Update `start_q_idx` to choose the start queue index to receive n/w packets. This should match the queue id used in lports section of [cndp_upf_1worker.jsonc](./conf/cndp_upf_1worker.jsonc).To get better performance (optional step), assign `cpuset-cpus` in [docker_setup.sh](./scripts/docker_setup.sh) to cores ids same as queue ids used to receive n/w packets.
+
+```
+diff --git a/conf/cndp_upf_1worker.jsonc b/conf/cndp_upf_1worker.jsonc
+index da60d51..b1c3df6 100644
+--- a/conf/cndp_upf_1worker.jsonc
++++ b/conf/cndp_upf_1worker.jsonc
+@@ -86,7 +86,7 @@
+     "lports": {
+         "enp134s0:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 6,
+             "umem": "umem0",
+             "region": 0,
+             "busy_poll": true,
+@@ -96,7 +96,7 @@
+         },
+         "enp136s0:0": {
```

Comment:
I think this should also be updated, accordingly, correct?

Reply:
What should be updated here ?

Reply:
The interface name (`enp136s0`). We can match the name to the "original" name that the interfaces that are mentioned in the upf.json file (i.e., `ens803f2` for the access and `ens803f3` for the core interface)